### DISCUSSION TITLE
When I use setState in the onchange event, the checkbox function is abnormal.

### DISCUSSION THREAD

**Discussion by yxd94772** [dowjones/react-dropdown-tree-select#587] [Q&A]
I want to save a copy of the data in selectedNodes, but when I use it in OnChange, an exception occurs. The check box is selected incorrectly

**Comment by gandhis1** [comment]
Need more detail and actual code. Can you prepare a code sandbox?

**Reply by yxd94772** [reply]
> Need more detail and actual code. Can you prepare a code sandbox?

https://codesandbox.io/s/zealous-brook-q7fonz?file=/src/App.js   
The checkbox dropdown of the selected subset will be closed，It doesn't look normal .

**Reply by yxd94772** [reply]
> Need more detail and actual code. Can you prepare a code sandbox?

I just want to keep the dropdown part because of the style problem, I need to listen for changing data and display it, because of this problem, I tried to solve this problem with HOC mentioned in this document with shouldComponentUpdate, but that would disable setState, I'm going to add and clear all the checkboxes, and there's nothing else I can do.

**Reply by gandhis1** [reply]
Yep, also you can use the `key` property on the `DropdownTreeSelect` component to control rendering
